src/bin/lpython.cpp

```diff
@@ -1526,32 +1506,52 @@ int main(int argc, char *argv[])
 
         if (endswith(arg_file, ".py"))
         {
+            int err = 0;
             if (backend == Backend::x86) {
-                return compile_to_binary_x86(arg_file, outfile,
+                err = compile_to_binary_x86(arg_file, outfile,
                         runtime_library_dir, compiler_options, time_report);
             } else if (backend == Backend::wasm) {
-                return compile_to_binary_wasm(arg_file, outfile,
+                err = compile_to_binary_wasm(arg_file, outfile,
                         runtime_library_dir, compiler_options, time_report);
             } else if (backend == Backend::wasm_x86) {
-                return compile_to_binary_wasm_to_x86(arg_file, outfile,
+                err = compile_to_binary_wasm_to_x86(arg_file, outfile,
                         runtime_library_dir, compiler_options, time_report);
-            }
-
-            std::string tmp_o = outfile + ".tmp.o";
-            int err;
-            if (backend == Backend::llvm) {
+            } else if (backend == Backend::llvm) {
 #ifdef HAVE_LFORTRAN_LLVM
-                err = compile_python_to_object_file(arg_file, tmp_o, runtime_library_dir, lpython_pass_manager, compiler_options, time_report);
+                std::string tmp_o = outfile + ".tmp.o";
+                err = compile_python_to_object_file(arg_file, tmp_o, runtime_library_dir,
+                    lpython_pass_manager, compiler_options, time_report);
+                if (err != 0) return err;
+                err = link_executable({tmp_o}, outfile, runtime_library_dir,
+                    backend, static_link, true, compiler_options);
+                if (err != 0) return err;
 #else
                 std::cerr << "Compiling Python files to object files requires the LLVM backend to be enabled. Recompile with `WITH_LLVM=yes`." << std::endl;
                 return 1;
 #endif
             } else {
                 throw LFortran::LCompilersException("Unsupported backend.");
             }
-            if (err) return err;
-            return link_executable({tmp_o}, outfile, runtime_library_dir,
-                    backend, static_link, true, compiler_options);
+
+            if (compiler_options.arg_o == "") {
+                if (backend == Backend::wasm) {
+                    err = system(("js " + outfile +".js").c_str());
+                } else {
+                    if (compiler_options.platform == LFortran::Platform::Windows) {
+                        err = system(outfile.c_str());
+                    } else {
+                        err = system(("./" + outfile).c_str());
+                    }
+                }
+                if (err != 0) {
+                    if (0 < err && err < 256) {
+                        return err;
+                    } else {
+                        return 1;
+                    }
+                }
```

Comment:
This is separate topic:
I recently came across something like:
```py
def test():
	quit(10)

test()
```
```console
$ lpython a.py
err = 2560
STOP
$ echo $?
1
```
So, I thought to replace this with the following change:
```suggestion
                if (err != 0) {
                    if (0 < err && err < 256) {
                        return err;
                    } if else (err % 256 == 0) {
                    	return err / 256;
                    } else {
                        return 1;
                    }
                }
```
@certik